explorer-ui/src/components/MatchGrid/MatchGrid.jsx

```diff
@@ -1,27 +1,111 @@
-import * as React from "react"
-import "./MatchGrid.css"
-import MatchCard from "../MatchCard/MatchCard"
-export default function MatchGrid({matches}) {
-    return (
-        <div className="match-grid">
-            <h1>Here are your matches!</h1>
-            {matches != null
-            ? matches.map((match, idx) => {
-                return (
-                    <MatchCard name="Sophia" bio={match.bio} picture={match.picture} country={match.country} accomodation={match.accomodation} />
-                // <div className="match" key={idx}>
-
-                //     <h2 className="matchNumber">Match #{idx}</h2>
-                //     <h2 className="macthPhoto">{match.picture}</h2>
-                //     <h2 className="macthBio">Bio: {match.bio}</h2>
-                //     <h2 className="macthDestination">Destination: {match.country}</h2>
-                //     <h2 className="macthAccomodation">Accomodation: {match.accomodation}</h2>
-                //     </div>
-
-                )
-            })
-            :<></>
+import * as React from "react";
+import "./MatchGrid.css";
+import MatchCard from "../MatchGrid/MatchCard/MatchCard";
+import MapContainer from "../Mapping/MapContainer";
+import { useEffect, useState } from "react";
+import Geocode from "react-geocode";
+
+import { useLoadScript } from "@react-google-maps/api";
+import ReactDOM from "react-dom/client";
+Geocode.setApiKey("AIzaSyA4B7q2I3Alla6f8udR0Nr-_3vB8lW5Te0");
+Geocode.setLanguage("en");
+Geocode.setLocationType("ROOFTOP");
+// Enable or disable logs. Its optional.
+Geocode.enableDebug();
+import { Map, GoogleApiWrapper, InfoWindow, Marker } from "google-maps-react";
+// const render = (status) => {
+//   return <h1>{status}</h1>;
+// };
+export default function MatchGrid({ profiles, userProfile }) {
+  const [matchesPlusCoordinates, setMatchesPlusCoordinates] = useState([]);
+  const [matches, setMatches] = useState([]);
+  const userProfiles = userProfile
+
+
+  useEffect(() => {
+    getMatches(profiles);
+}, [])
+  // TODO: Create a function that filters through the list of users to get the matches of this specific user
+  const getMatches = () => {
+    for (let i = 0; i < profiles.length; i++) {
+      if (
+        profiles[i].country == userProfile.country 
+        // &&
+        // profiles[i].travelMonth == userProfile.travelMonth &&
+        // profiles[i].accomodation == userProfile.accomodation
+      ) {
+        setMatches([...matches, profiles[i]])
+      }
```

Comment:
capture all changes then set state outside of fxn

Reply:
Changed to creating a local array "profilesToAdd", appending each profile to it and then calling setMatches([...matches, ...profilesToAdd]). Is this what you meant?